x-pack/plugins/cases/server/services/attachments/index.ts

```diff
@@ -364,7 +411,9 @@ export class AttachmentService {
            */
           references:
             refsWithExternalRefId.length > 0 || didDeleteOperation
-              ? refsWithExternalRefId
+              ? uniqBy([...refsWithExternalRefId, ...extractedReferences], 'id')
+              : extractedReferences.length > 0
```

Comment:
`extract` do not remove references. They extract the references from the attributes. If they are empty it means that the are no references to add from the attributes.